subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -775,10 +782,50 @@ static void mfy_sync_offset_get(void *param)
 	/* NOTE: as remainder not used in scheduling primary PDU
 	 * packet timer starts transmission after 1 tick hence the +1.
 	 */
-	sync->lll.ticks_offset = ticks_to_expire + 1;
+	lll_sync->ticks_offset = ticks_to_expire + 1;
 
 	pdu = lll_adv_aux_data_curr_get(adv->lll.aux);
-	lll_adv_sync_offset_fill(ticks_to_expire, 0, pdu);
+	si = sync_info_get(pdu);
+	sync_info_offset_fill(si, ticks_to_expire, 0);
+	si->evt_cntr = lll_sync->event_counter + lll_sync->latency_prepare;
+}
+
+static inline struct pdu_adv_sync_info *sync_info_get(struct pdu_adv *pdu)
+{
+	struct pdu_adv_com_ext_adv *p;
+	struct pdu_adv_hdr *h;
+	uint8_t *ptr;
+
+	p = (void *)&pdu->adv_ext_ind;
+	h = (void *)p->ext_hdr_adi_adv_data;
+	ptr = (uint8_t *)h + sizeof(*h);
+
+	if (h->adv_addr) {
+		ptr += BDADDR_SIZE;
+	}
+
+	if (h->adi) {
+		ptr += sizeof(struct pdu_adv_adi);
+	}
+
+	if (h->aux_ptr) {
+		ptr += sizeof(struct pdu_adv_aux_ptr);
+	}
+
+	return (void *)ptr;
+}
+
+static inline void sync_info_offset_fill(struct pdu_adv_sync_info *si,
+					 uint32_t ticks_offset,
+					 uint32_t start_us)
+{
+	if (si->offs_units) {
+		si->offs = (HAL_TICKER_TICKS_TO_US(ticks_offset) - start_us) /
```

Comment:
Pulling up "(HAL_TICKER_TICKS_TO_US(ticks_offset) - start_us)" into a variable above the if/else should save a little code space.

Reply:
offs is 13-bits, in order not to loose data, an auto variable would be required before the division, hence the conscious decision to keep it the way it is, and let the compiler optimize the correct way.

Reply:
I believe that an auto variable would be cleaner and more efficient, but it's not something I am willing to spend time on discussing :-)

Reply:
I implemented your suggestion, used a uint32_t variable which should be able to hold a 13-bit value times 300 us. And I got a 8 byte reduction in code space on CM4.